File: mobile/src/main/java/packi/day/di/FocusCelebrationViewModelProvider.java
package packi.day.di;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;

import dagger.Module;
import dagger.Provides;
import packi.day.main.FocusCelebrationViewModel;
import packi.day.store.StoreLocator;

@Module
public class FocusCelebrationViewModelProvider {

    @Provides
    ViewModelProvider.Factory provideFocusCelebrationViewModel(StoreLocator locator) {
        return new ViewModelProvider.Factory() {
            @NonNull
            @Override
            public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
                return (T) new FocusCelebrationViewModel(locator.getStore(), null);
            }
        };
    }

}

File: mainscreen/src/main/java/packi/day/di/FocusCelebrationViewModelProvider.kt
@file:Suppress("UNCHECKED_CAST")

package packi.day.di

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import dagger.Module
import dagger.Provides
import packi.day.main.FocusCelebrationViewModel
import packi.day.store.StoreLocator

@Module
object FocusCelebrationViewModelProvider {

    @Provides
    fun provideFocusCelebrationViewModel(locator: StoreLocator): FocusCelebrationViewModel {
        return object : ViewModelProvider.Factory {
            override fun <T : ViewModel> create(modelClass: Class<T>): T {
                return FocusCelebrationViewModel(locator.getStore()) as T
            }
        }.create(FocusCelebrationViewModel::class.java)
    }
}